packages/filebrowser-extension/src/index.ts

```diff
@@ -541,6 +542,20 @@ function addCommands(
       return Promise.all(
         toArray(
           map(widget.selectedItems(), item => {
+            // check for nonstandard content type
+            const ft = widget.model.manager.registry.getFileTypeForModel(item);
+            if (ft && ft.contentType === 'dirlike') {
+              // cast to FileBrowser-like
+              const browserlike = (ft.browser || widget) as SideBarWidget & {
+                model: FileBrowserModel;
+              };
```

Comment:
How about, if instead of adding a `browser` attribute, we use `Private.navigateToPath()` ? We might have to update `Private.getBrowserForPath()` to automatically create any missing browsers:

```ts
if (!browserForPath) {
  return factory.createFileBrowser(...);
}
```

Reply:
Using `navigateToPath` was the original design, but I changed it so that it matches the code I added to `filebrowser/src/listing.ts`. This ensures that opening by context menu (implemented here in `filebrowser-extension/src/index.ts`) and opening by double click (implemented in `filebrowser/src/listing.ts`) use the same code. `navigateToPath` ultimately relies on the `FileBrowser` tracker supplied by `FileBrowserFactory`, so there's no practical way to use anything like `navigateToPath` in `filebrowser/src/listing.ts`.

Ideally double clicking and "open" in context menu should use the same code path. I'm currently thinking through the best way to make that happen.

Creating missing browsers sounds like an interesting idea, but I don't quite get it (in terms of the current `FileBrowserFactory` infrastructure and the like). Can you say more about that?